README.md

```diff
@@ -1,568 +1,181 @@
-
-```python
-config.serialize("model_name", data, "prompt_name")
-```
-
-### Prompt `run`
-
-`run` is used to run inference for the specified `Prompt`.
-
-```python
-config.run("prompt_name", params)
-```
-
-### `run_with_dependencies`
-
-This is a variant of `run` -- this re-runs all prompt dependencies.
-For example, in [`travel.aiconfig.json`](#download-travelaiconfigjson), the `gen_itinerary` prompt references the output of the `get_activities` prompt using `{{get_activities.output}}`.
-
-Running this function will first execute `get_activities`, and use its output to resolve the `gen_itinerary` prompt before executing it.
-This is transitive, so it computes the Directed Acyclic Graph of dependencies to execute. Complex relationships can be modeled this way.
-
-```python
-config.run_with_dependencies("gen_itinerary")
+# first, setup your openai key: https://platform.openai.com/api-keys
+# in your CLI, set the environment variable
+export OPENAI_API_KEY=my_key
 ```
 
-### Updating metadata and parameters
-
-Use the `get/set_metadata` and `get/set_parameter` methods to interact with metadata and parameters (`set_parameter` is just syntactic sugar to update `"metadata.parameters"`)
-
 ```python
-config.set_metadata("key", data, "prompt_name")
-```
-
-Note: if `"prompt_name"` is specified, the metadata is updated specifically for that prompt. Otherwise, the global metadata is updated.
-
-### Register new `ModelParser`
-
-Use the `AIConfigRuntime.register_model_parser` if you want to use a different `ModelParser`, or configure AIConfig to work with an additional model.
-
-AIConfig uses the model name string to retrieve the right `ModelParser` for a given Prompt (see `AIConfigRuntime.get_model_parser`), so you can register a different ModelParser for the same ID to override which `ModelParser` handles a Prompt.
-
-For example, suppose I want to use `MyOpenAIModelParser` to handle `gpt-4` prompts. I can do the following at the start of my application:
+# load your AIConfig
+from aiconfig import AIConfigRuntime, InferenceOptions
 
-```python
-AIConfigRuntime.register_model_parser(myModelParserInstance, ["gpt-4"])
-```
+config = AIConfigRuntime.load("trip_planner_aiconfig.json")
 
-### Callback events
+# setup streaming
+inference_options = InferenceOptions(stream=True)
 
-Use callback events to trace and monitor what's going on -- helpful for debugging and observability.
+# run a prompt
+# `get_activities` prompt generates a list of activities in specified city ('london' is default)
+get_activities_response = await config.run("get_activities", options=inference_options)
 
-```python
-from aiconfig import AIConfigRuntime, CallbackEvent, CallbackManager
-config = AIConfigRuntime.load('aiconfig.json')
+# run a prompt with a different parameter
+# update city parameter to 'san francisco'
+params = {“city” : “san francisco”}
```

Comment:
Inline this in the line below. i.e. config.run(params={...})

Same for all examples below